pkg/node/manager/manager.go

```diff
@@ -90,14 +93,47 @@ type Manager struct {
 	metricDatapathValidations prometheus.Counter
 }
 
+// Subscribe subscribes the given node handler to node events.
+func (m *Manager) Subscribe(nh datapath.NodeHandler) {
+	m.mutex.RLock()
+	defer m.mutex.RUnlock()
+	m.nodeHandlersMu.Lock()
+	m.nodeHandlers[nh] = struct{}{}
+	m.nodeHandlersMu.Unlock()
+	// Add all nodes already received by the manager.
+	for _, v := range m.nodes {
+		v.mutex.Lock()
+		nh.NodeAdd(v.node)
+		v.mutex.Unlock()
+	}
+}
+
+// Unsubscribe unsubscribes the given node handler with node events.
+func (m *Manager) Unsubscribe(nh datapath.NodeHandler) {
+	m.nodeHandlersMu.Lock()
```

Comment:
`Subscribe` grabs `m.mutex` first, then `m.nodeHandlersMu`; Is this also necessary for unsubscribe?

Reply:
[1] I don't understand what happened here. `m.nodes` does not need the `m.mutex` so I've removed it.